scapy/utils.py

```diff
@@ -1171,11 +1171,17 @@ def __call__(cls, filename):  # type: ignore
         raise Scapy_Exception("Not a supported capture file")
 
     @staticmethod
-    def open(fname  # type: Union[IO[bytes], str]
+    def open(fname  # type: Union[IO[bytes], str, bytes, os.PathLike[Any]]
              ):
-        # type: (...) -> Tuple[str, _ByteStream, bytes]
+        # type: (...) -> Tuple[Union[IO[bytes], str, bytes, os.PathLike[Any]], _ByteStream, bytes]
         """Open (if necessary) filename, and read the magic."""
-        if isinstance(fname, str):
+        try:
+            # Allow os.PathLike object when running on Python 3.4+
+            allowed_fname_types = (str, bytes, os.PathLike)
+        except AttributeError:
+            allowed_fname_types = (str, bytes)  # type: ignore 
```

Comment:
```suggestion
            allowed_fname_types = (str, bytes)  # type: ignore
```
The trailing space will cause tests failure.